src/connectors/menu/connectMenu.js

```diff
@@ -56,13 +61,34 @@ export default function connectMenu(renderFn) {
       attributeName,
       limit = 10,
       sortBy = ['isRefined', 'count:desc'],
+      showMoreLimit = 0,
     } = widgetParams;
 
     if (!attributeName) {
       throw new Error(usage);
     }
 
     return {
+      isShowingMore: false,
+
+      // Provide the same function to the `renderFn` so that way the user
+      // has to only bind it once when `isFirstRendering` for instance
+      toggleShowMore() {},
+      cachedToggleShowMore() { this.toggleShowMore(); },
+
+      createToggleShowMore({results, instantSearchInstance}) {
+        return () => {
+          this.isShowingMore = !this.isShowingMore;
+          this.render({results, instantSearchInstance});
+        };
+      },
+
+      getLimit() {
+        return showMoreLimit > limit && this.isShowingMore
```

Comment:
Yes, I think it's the best solution: checking the parameters as we receive them.